text/0000-transaction-payload/0000-transaction-payload.md

```diff
@@ -0,0 +1,748 @@
+This block signs the entire <i>Transaction Essence</i> part of a <i>Transaction Payload</i> including the optional payload.
+
+##### Reference Unlock block
+
+<details>
+    <summary>Format</summary>
+    <table>
+        <tr>
+            <th>Name</th>
+            <th>Type</th>
+            <th>Description</th>
+        </tr>
+        <tr>
+            <td>Unlock Type</td>
+            <td>uint8</td>
+            <td>
+                Set to <strong>value 1</strong> to denote a <i>Reference Unlock Block</i>.
+            </td>
+        </tr>
+        <tr>
+            <td>Reference</td>
+            <td>uint16</td>
+            <td>Represents the index of a previous unlock block.</td>
+        </tr>
+    </table>
+</details>
+
+A <i>Reference Unlock Block</i> defines an <i>Unlock Block</i> which references a previous <i>Unlock Block</i> (which must not be another <i>Reference Unlock Block</i>). It must be used if multiple inputs can be unlocked through the same origin <i>Unlock Block</i>.
+
+Example:
+Consider an <i>Transaction Essence</i> containing <i>UTXO Inputs</i> A, B and C, where A and C are both spending the UTXOs originating from the same Ed25519 address. The <i>Unlock Block</i> part must thereby have following structure:
+
+| Index | Must Contain                                                                                                         |
+| ----- | ---------------------------------------------------------------------------------------------------------------- |
+| 0     | A <i>Signature Unlock Block</i> holding the corresponding Ed25519 signature to unlock A and C.                    |
+| 1     | A <i>Signature Unlock Block</i> which unlocks B.                                                                 |
+| 2     | A <i>Reference Unlock Block</i> which references index 0, since C also gets unlocked by the same signature as A. |
+
+## Validation
+
+A <i>Transaction Payload</i> payload has different validation stages, since some validation steps can only be executed at the point when certain information has (or has not) been received. We therefore distinguish between syntactical- and semantic validation.
+
+### Syntactical Validation
+
+This validation can commence as soon as the transaction data has been received in its entirety. It validates the structure but not the signatures of the transaction. If the transaction does not pass this stage, it must not be broadcasted further and can be discarded right away.
+
+The following criteria defines whether the transaction passes the syntactical validation:
+* `Transaction Essence Type` value must be 0, denoting an `Transaction Essence`.
+* Inputs:
+    * `Inputs Count` must be 0 < x < 127.
+    * At least one input must be specified.
+    * `Input Type` value must be 0, denoting an `UTXO Input`.
+    * `UTXO Input`:
+        * `Transaction Output Index` must be 0 ≤ x < 127.
```

Comment:
Input/output count is `0 < x < 127` -> 126 possible.
Output index is `0 ≤ x < 127` -> 127 possible.